src/app/modular/post/actions.ts

```diff
@@ -39,3 +42,38 @@ ThunkAction<void, RootState, null, any> => (async (dispatch) => {
     console.error(err);
   }
 });
+
+export const createPost = ({
+  title,
+  description,
+  desirements,
+  requirements,
+}: {
+  title: string,
+  description: string,
+  desirements: string[],
+  requirements: string[],
+}): ThunkAction<void, RootState, null, any> => (async (dispatch) => {
+  dispatch(loadingStart());
+  try {
+    const res: any = await request.post({
+      url: 'posts',
+      body: {
+        title,
+        description,
+        desirements,
+        requirements,
+      },
+    });
+    res.updatedAt = new Date(res.updatedAt);
+    res.createdAt = new Date(res.createdAt);
+    res.expiresAt = new Date(res.expiresAt);
+    res.author.updatedAt = new Date(res.author.updatedAt);
+    dispatch(addPost(res));
+    dispatch(loadingSuccess());
```

Comment:
`addPost(res)` should actually trigger stop loading so we don't need to separately dispatch a `loadingSuccess()` action